backend/src/main/java/com/huellapositiva/domain/actions/FetchESALAction.java

```diff
@@ -0,0 +1,40 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.application.dto.GetESALResponseDto;
+import com.huellapositiva.application.exception.UserNotFoundException;
+import com.huellapositiva.infrastructure.orm.entities.JpaContactPerson;
+import com.huellapositiva.infrastructure.orm.entities.JpaESAL;
+import com.huellapositiva.infrastructure.orm.entities.JpaLocation;
+import com.huellapositiva.infrastructure.orm.repository.JpaContactPersonRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FetchESALAction {
+
+    @Autowired
+    private JpaContactPersonRepository jpaContactPersonRepository;
+
+    public GetESALResponseDto execute(String accountId) {
+        JpaContactPerson jpaContactPerson = jpaContactPersonRepository.findByAccountIdWithProfile(accountId)
+                .orElseThrow(() -> new UserNotFoundException("Contact Person not found. Account ID: " + accountId));
+        JpaESAL joinedEsal = jpaContactPerson.getJoinedEsal();
+
+        GetESALResponseDto.GetESALResponseDtoBuilder builder = GetESALResponseDto.builder();
+        builder.name(joinedEsal.getName())
```

Comment:
I think this might lead to a NullPointerException in some cases.

I see two flows for creating an ESAL:

1. Flow (no issue)
- Create contact person
- Contact person creates an ESAL and both are linked

2. Flow (multiple issue)
2.1 Reviser creates an ESAL which leads to having an ESAL witthout a ContactPerson
2.2 If you retrieve the ESAL you get a NPE, also there is no apparent way to link a ContactPerson to an ESAL that was created by a reviser.

Reply:
Solved, now it receives the ID of an ESAL and we check if the logged contact person is its owner, it doesn't throw NPE anymore.